src/client/datascience/jupyter/jupyterImporter.ts

```diff
@@ -105,7 +105,9 @@ export class JupyterImporter implements INotebookImporter {
     };
 
     private get defaultCellMarker(): string {
-        return this.configuration.getSettings().defaultCellMarker || Identifiers.DefaultCodeCellMarker;
+        const language = window.activeTextEditor?.document.languageId
```

Comment:
We actually have an abstraction around the VScode 'window' namepace called ```IDocumentManager```. You'd inject an ```IDocumentManager``` into the constructor of the importer and use that here instead of the window namespace.

This abstraction allows us to have tests that can run without VS code being involved.

Reply:
In this case, do I call the active text editor: 

```
const language = this.DocumentManager.activeTextEditor?.document.languageId
```

where the constructor now has: 

```
@inject(IDocumentManager) private readonly DocumentManager: IDocumentManager
```

Reply:
Yes something like that. The importer may be called for things other than the active document though. It would be good to check where defaultCellMarker is used (under what circumstances). The caller of defaultCellMarker may have a file.

Reply:
Where can I find the caller of defaultCellMarker?

Reply:
If you have typescript reference tracking turned on, the IDE can tell you:

![image](https://user-images.githubusercontent.com/19672699/108541513-641a3780-7297-11eb-8012-f862f4873202.png)

Or you can search for defaultCellMarker in that same file. It's a private member.